### ISSUE TITLE
Warn about dead code after exit

### ISSUE THREAD

**Issue by walles** [koalaman/shellcheck#1031]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

Depending on your view of the world, #760 might be related.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
echo foo
exit
echo bar
```

#### Here's what shellcheck currently says:
Nothing

#### Here's what I wanted or expected to see:
A warning that the `echo bar` line will never be executed.

**Comment by nihilus** [comment]
Great suggestion however just finding the last exit-statement and check for code after it won't help if the exit is inside a conditional statement, which must be considered when implementing this.

**Comment by nihilus** [comment]
I found this https://bobcopeland.com/blog/2012/10/goto-in-bash/ which would yield in a false positive if used, kinda neat hack however.
